sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/DiagnosticVerifier.cs

```diff
@@ -151,17 +151,12 @@ private static void DumpActualDiagnostics(string languageVersion, IEnumerable<Di
             bool verifyNoException = true,
             CancellationToken? cancellationToken = null)
         {
-            var compilationOptions = compilation.Language == LanguageNames.CSharp
```

Comment:
Bad things:
* The input `compilation.Options` were completely ignored.
* All tests were actually build with default `OutputKind.ConsoleApplication` causing most of them containing build errors because of missing `Main` entry point.
* This `GetAllDiagnostics` did a workaround that hides those build errors for the next `VerifyBuildErrors` step

New solution:
I moved the orchestration (`DynamicallyLinkedLibrary` and `allowUnsafe`) directly to the `AddProject` so now it's default for everything we use.